### DISCUSSION TITLE
Table does not update, when clearing

### DISCUSSION THREAD

**Discussion by Shahin-rmz** [zauberzeug/nicegui#864] [Q&A] [answered]
### Question

Hello, I made an updatable table. 
when adding items to it it works smoothly without problem.
Here is the function of `ui.table`.
```python
def on_submit():
    data_dic = {'name': name_item.value, 'count': count_item.value, 'price':
                price_item.value, "sum_unit":
                (float(count_item.value)*float(price_item.value))}
    rows.append(data_dic)
    table.update()
    name_item.set_value("")
    count_item.set_value("")
    price_item.set_value("")
    sum_total.append(data_dic["sum_unit"])
```
[Screencast from 29.04.2023 21:56:56.webm](https://user-images.githubusercontent.com/32805117/235321845-713c80c9-8332-4186-88c6-d1c80381ce47.webm)

The problem is by clearing.
When I clear, I get no error, and although `table.update()` will be triggered, but it won't remove anything in front end.
(As you can see, back end hast been successfully updated, the `list` is empty.)
why is that?
here is the `clearing function`

```python
def clear_table():
    global rows, sum_total
    rows = []
    sum_total = []
    table.update()
```

**Comment by falkoschindler** [comment, marked as answer]
With `rows = []` your not changing the list referenced by `rows`, but you assign a *new* list to `rows`. This new list has nothing to to with the table. This can be avoided using `rows.clear()` or `rows[:] = []`. Same for `sum_total`.

**Reply by Shahin-rmz** [reply]
Works like a charm.

PS.
 :clock1: :eyes: 
:)
